server/src/server.common.ts

```diff
@@ -152,6 +185,51 @@ export function startServer(connection: _Connection) {
 		}
 	}
 
+	connection.onCodeAction((params: CodeActionParams) => {
+		const diagnostics = params.context.diagnostics;
+
+		const codeActions: CodeAction[] = [];
+
+		for(const d of diagnostics) {
+			if (d.code) {
+				const fix = lookupFixes(params.textDocument.uri, d);
+				if (fix) {
+					codeActions.push(fix);
+				}
+			}
+		}
+
+		return codeActions;
+	});
+
+	function lookupFixes(docUri: DocumentUri, diagnostic: Diagnostic): CodeAction | undefined {
+		// If doc doesnt exist, we can exit now.
+		const doc = documents.get(docUri);
+		if (!doc) {
+			return;
+		}
+		
+		// Determine indent
+		let indent = "\t";
+		if (insertSpaces) {
+			indent = " ";
+			for (let i = 1; i < tabSize; i++) {
+				indent += " ";
+			}
+		}
```

Comment:
Just realized in testing that indents are either 2 spaces or a tab. I will remove this logic.

I will default to 2 spaces, or a single tab based on settings.